### DISCUSSION TITLE
Using spk2utt & utt2spk in TTS task with x_vector

### DISCUSSION THREAD

**Discussion by pratikkumar018** [espnet/espnet#4703] [Q&A] [answered]
I was checking this https://espnet.github.io/espnet/notebook/tts_cli.html 
It says 
"In the ESPnet, speaker information is not used for any processing.
Therefore, utt2spk and spk2utt can be a dummy."

I want to ask when we are using x_vector(Libritts recipe) & set use_xvector to true, are we still not using spk2utt & utt2spk files & using only x_vector net(xvector_nnet_1a) to get the spembs or we are using data of these two files as well?

**Comment by kan-bayashi** [comment, marked as answer]
The notebook is for ESPnet1, now ESPnet2 utilizes `utt2spk` and `spk2utt` files for multi-speaker case, especially using speaker ID as the input (`use_sid=true`).
If you use `xvector`, as a default, we only use utterance-wise xvector so the information of `utt2spk` is not used.
But if you want to use speaker-averaged xvector (`spk_xvector.scp`) in the inference, `utt2spk` may be needed since `spk_xvector.scp` is created using `spk2utt`.

**Reply by pratikkumar018** [reply]
Thanks for the reply. So , what I understood is if we set use_sid=true, it will use utt2spk file . 
But if we set use_sid=false & set use_xvector=true, it will not use utt2spk file, it will treat every utterance as independent utterance & get the xvector for that. Please correct me if I am wrong.

**Reply by kan-bayashi** [reply]
> it will treat every utterance as independent utterance & get the xvector for that. Please correct me if I am wrong.

`xvector.scp` and `spk_xvector.scp` are created.
And `spk_xvector.scp` used `spk2utt` information.
`spk_xvector.scp` is not used as a default in the training.

**Reply by pratikkumar018** [reply]
Got it. Thanks

**Comment by pratikkumar018** [comment]
@kan-bayashi 
Hi Can you please help how to configure using spk_xvector.scp in training instead of xvector.scp
